### ISSUE TITLE
port twisted.names.server to python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8195]
|<img alt="bluec0re's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| bluec0re reported|
|-|-|
|Trac ID|trac#8195|
|Type|enhancement|
|Created|2016-02-03 08:28:29Z|
|Branch|https://github.com/twisted/twisted/tree/names-server-py3-8195-2|

See previous tickets: [#6093](https://github.com/twisted/twisted/issues/6093) [#6056](https://github.com/twisted/twisted/issues/6056) [#6059](https://github.com/twisted/twisted/issues/6059) [#6057](https://github.com/twisted/twisted/issues/6057) [#6058](https://github.com/twisted/twisted/issues/6058) [#6092](https://github.com/twisted/twisted/issues/6092)

Attachments:

* [o](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d30/d3063b6a09e5da6ff5745bb4bdf557c84e141f1a/7a81af3e591ac713f81ea1efe93dcf36157d8376) (930 bytes) - added by pawelmhm on 2016-03-12 15:05:12Z - 
* [patch.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d30/d3063b6a09e5da6ff5745bb4bdf557c84e141f1a/9c2db015e47d179fd5565f2f2144ace7e021399c.diff) (1890 bytes) - added by pawelmhm on 2016-03-19 18:34:44Z - 
* [path.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d30/d3063b6a09e5da6ff5745bb4bdf557c84e141f1a/2d96ea0eb6ac138798a38b526dbab79dc1e42b57.diff) (3166 bytes) - added by pawelmhm on 2016-03-20 18:34:00Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__8195 8195
type__enhancement enhancement
reporter__bluec0re bluec0re
priority__normal normal
milestone__None None
branch__branches_names_server_py3_8195_2 branches/names-server-py3-8195-2
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__names names
keywords__None None
time__1454488109341273 1454488109341273
changetime__1459204604173872 1459204604173872
version__None None
owner__pawelmhm pawelmhm

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

I've added this to [Plan/Python3](https://twistedmatrix.com/trac/wiki/Plan/Python3).

**Comment by twisted-trac** [comment]
|[<img alt="pawelmhm's avatar" src="https://avatars.githubusercontent.com/u/2700942?s=50" width="50" height="50">](https://github.com/pawelmhm)<a name="note_2"></a>|@pawelmhm set owner to @adiroiban|
|-|-|

this is actually pretty tiny change, here's the patch.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @pawelmhm|
|-|-|

Thanks for the patch.

Please also check all the tests related to  twisted.names.server and also port them (if required) and include them in the python3 test suite. See dist3.py

-------

Also check http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles 

Your patch requires a news fragment to advertise the new capability as part of the next release notes.

--------

Looking forward for the new patch and have the new tests included into our python3 test suite.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="pawelmhm's avatar" src="https://avatars.githubusercontent.com/u/2700942?s=50" width="50" height="50">](https://github.com/pawelmhm)<a name="note_4"></a>|@pawelmhm set owner to @adiroiban|
|-|-|

tests seem to pass just fine without changes, I updated dist3.py module list and added news file

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban commented|
|-|-|

(In [47030]) Branching to names-server-py3-8195.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban set owner to @pawelmhm|
|-|-|

Thanks for the patch!

------

Please note that gotResolverResponse docstring should be updated... it was wrong before but now that we have touched it, it should be a good time to fix the docstring.

I have never used twisted.names so I can not suggest another name for that argument.

--------

To have a closer behaviour between py2.7 and py3 all ported files should do the `from __futute__ import` for absolute import / division and what other things can be imported :)

I have aleady commted that changes.

------

server.py advertises that its tests are in

```
# -*- test-case-name: twisted.names.test.test_names,twisted.names.test.test_server -*-
```

Since the patch is small, can we have twisted.names.test.test_names also enabled with this patch?

------ 


I have applied the patch and send it to buildbot.

Please check my comments and send your feedback.

If possible, please send a patch on top of the branch dedicated to this ticket.

Many thanks for your contribution!

**Comment by twisted-trac** [comment]
|[<img alt="pawelmhm's avatar" src="https://avatars.githubusercontent.com/u/2700942?s=50" width="50" height="50">](https://github.com/pawelmhm)<a name="note_7"></a>|@pawelmhm set owner to @adiroiban|
|-|-|

> Since the patch is small, can we have twisted.names.test.test_names also enabled with this patch?

unfortunately tests for test_names are failing because of unported authority.py. Should this be ported in this ticket or in new one? I see there is convention of one names module per ticket so maybe would be good to keep this convention?

**Comment by twisted-trac** [comment]
|[<img alt="pawelmhm's avatar" src="https://avatars.githubusercontent.com/u/2700942?s=50" width="50" height="50">](https://github.com/pawelmhm)<a name="note_8"></a>|@pawelmhm commented|
|-|-|

is there anything else to do here?

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban commented|
|-|-|

Hi,

Sorry for the confusion. You attached a patch without any comment about the patch and I was not sure if more comments will follow.

Is ok to port authority.py in a separate ticket. Please create a ticket and reference it from here.

Will apply the patch and see how it goes.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_10"></a>|@adiroiban commented|
|-|-|

(In [47094]) Branching to names-server-py3-8195-2.

**Comment by twisted-trac** [comment]
|[<img alt="pawelmhm's avatar" src="https://avatars.githubusercontent.com/u/2700942?s=50" width="50" height="50">](https://github.com/pawelmhm)<a name="note_11"></a>|@pawelmhm commented|
|-|-|

created ticket for authority here [#8259](https://github.com/twisted/twisted/issues/8259)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_12"></a>|@adiroiban commented|
|-|-|

Thanks.

Test look good.

There were a couple of minor issues....but hope they are fixed

ex

```
************* Module twisted.names.server
C0103:264,34:DNSServerFactory.gotResolverResponse: Invalid name "ans_auth_add" for type variable (should match ([a-z_][a-zA-Z0-9]*)$)
```

I have renamed `ans_auth_add` to `response` as I hope it is a better name :)

Waiting for the test results and if buildbot is happy and nobody has any other objection I will merge.

Looking forward for  [#8259](https://github.com/twisted/twisted/issues/8259)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_13"></a>|@adiroiban set owner to @pawelmhm|
|-|-|

Looks good. will merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_14"></a>|@adiroiban set status to `closed`|
|-|-|

(In [47098]) Merge names-server-py3-8195-2: Port twisted.names.server to Python 3.

Author: pawelmhm
Reviewer: adiroiban
Fixes: [#8195](https://github.com/twisted/twisted/issues/8195)
